File: ios/TxParamsRecord.swift
import ExpoModulesCore

struct TxParamsRecord: Record {
    @Field
    var to: String?
    @Field
    var value: String?
    @Field
    var data: String?
    @Field
    var delegateCall: Bool = false
}

extension TxParamsRecord {
    func verify() throws {
        if to == nil {
            throw Rtn4337Exception("rpcUrl is required")
        }
    }
}

File: android/src/main/java/io/cometh/rtn4337/types/TxParamsRecord.kt
package io.cometh.rtn4337.types

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class TxParamsRecord : Record {
    @Field
    val to: String? = null

    @Field
    val value: String? = null

    @Field
    val data: String? = null

    @Field
    val delegateCall: Boolean = false

}


fun TxParamsRecord.verifyParams() {
    requireNotNull(to) { "to is required" }
}